ui/plugins/eu.esdihumboldt.hale.ui.common/src/eu/esdihumboldt/hale/ui/common/definition/viewer/OptionalPropertiesFilter.java

```diff
@@ -89,16 +91,20 @@ private boolean areChildrenOptional(EntityDefinition entityDef) {
 						.getConstraint(Cardinality.class);
 				if (cardinality.getMinOccurs() != 0)
 					return false;
+				else if (!areChildrenOptional(entityDef))
+					return false;
 			}
 			else if (def instanceof PropertyDefinition) {
 				Cardinality cardinality = ((PropertyDefinition) def)
 						.getConstraint(Cardinality.class);
 				if (cardinality.getMinOccurs() != 0)
 					return false;
-			}
 
-			if (!areChildrenOptional(child))
-				return false;
+				if (((PropertyDefinition) def).getConstraint(NillableFlag.class).isEnabled()) {
```

Comment:
No else here as well? If it is not Nillable (and minOccurs is not zero, which is the only condition that Nillable needs to be checked anyway), then it is not optional.